coalahtml/_coalahtml/app/controllers/main.js

```diff
@@ -2,45 +2,47 @@
 
 angular.module('coalaHtmlApp')
   .controller('MainCtrl', function ($scope, $http) {
+    $http.get("data/Constants.json").success(function(content) {
+      $scope.Constants = content;
+      parseCoalaProject($scope.Constants.data);
+    });
     var parseCoalaProject = function(projectDir, coalaJSON) {
-      coalaJSON = coalaJSON || projectDir + '/coala.json';
-      $scope.data = [];
-      $http.get(coalaJSON).success(function(response){
-        var knownFiles = {};
-        $scope.content = {};
-
-        var parseResult = function(el){
-          el.affected_code.forEach(function(sourceRange){
-              if (!(sourceRange.file in knownFiles)){
-                  knownFiles[sourceRange.file] = [];
-
-                  $http.get(projectDir + "/" + sourceRange.file)
-                    .success(function(content) {
-                      $scope.content[sourceRange.file] = content.split("\n");
+      $http.get(projectDir + $scope.Constants.file_data)
+        .success(function(file_data) {
+            $scope.file_dict = {};
+            Object.keys(file_data).forEach(function(file_name) {
+              if (!(file_name in $scope.file_dict)) {
+                $scope.file_dict[file_name]=file_data[file_name];
+              }
+            });
+          coalaJSON = coalaJSON || projectDir + $scope.Constants.coala;
+          $scope.data = [];
+          $http.get(coalaJSON).success(function(response) {
+            var knownFiles = {};
+            var parseResult = function(el) {
+                el.affected_code.forEach(function(sourceRange){
+                    if (!(sourceRange.file in knownFiles)){
+                        knownFiles[sourceRange.file] = [];
+                    }
+                    knownFiles[sourceRange.file].push({
+                      "start":    sourceRange.start.line,
+                      "end":      sourceRange.end.line,
+                      "diffs":    el.diffs,
+                      "message":  el.message,
+                      "origin":   el.origin,
+                      "severity": el.severity
                     });
+                });
+            };
+            for (var section in response.results) {
+              if (response.results[section][0]){ // If error exists
```

Comment:
`response.results[section]` is an array, isn't it? So do you basically check here that the array is not empty?